ibis/backends/duckdb/datatypes.py

```diff
@@ -1,98 +1,94 @@
 from __future__ import annotations
 
 import duckdb_engine.datatypes as ducktypes
-import parsy
 import sqlalchemy.dialects.postgresql as psql
-import toolz
+import sqlglot as sg
+from sqlglot.expressions import DataType
 
+import ibis.common.exceptions as exc
 import ibis.expr.datatypes as dt
 from ibis.backends.base.sql.alchemy.datatypes import AlchemyType
-from ibis.common.parsing import (
-    COMMA,
-    FIELD,
-    LBRACKET,
-    LPAREN,
-    PRECISION,
-    RAW_STRING,
-    RBRACKET,
-    RPAREN,
-    SCALE,
-    spaceless,
-    spaceless_string,
-)
-
-
-def parse(text: str, default_decimal_parameters=(18, 3)) -> dt.DataType:
-    """Parse a DuckDB type into an ibis data type."""
-    primitive = (
-        spaceless_string("interval").result(dt.Interval("us"))
-        | spaceless_string("hugeint", "int128").result(dt.Decimal(38, 0))
-        | spaceless_string("bigint", "int8", "long").result(dt.int64)
-        | spaceless_string("boolean", "bool", "logical").result(dt.boolean)
-        | spaceless_string("blob", "bytea", "binary", "varbinary").result(dt.binary)
-        | spaceless_string("double", "float8").result(dt.float64)
-        | spaceless_string("real", "float4", "float").result(dt.float32)
-        | spaceless_string("smallint", "int2", "short").result(dt.int16)
-        | spaceless_string(
-            "timestamp with time zone", "timestamp_tz", "datetime"
-        ).result(dt.Timestamp(timezone="UTC"))
-        | spaceless_string("timestamp_sec", "timestamp_s").result(
-            dt.Timestamp(timezone="UTC", scale=0)
-        )
-        | spaceless_string("timestamp_ms").result(dt.Timestamp(timezone="UTC", scale=3))
-        | spaceless_string("timestamp_us").result(dt.Timestamp(timezone="UTC", scale=6))
-        | spaceless_string("timestamp_ns").result(dt.Timestamp(timezone="UTC", scale=9))
-        | spaceless_string("timestamp").result(dt.Timestamp(timezone="UTC"))
-        | spaceless_string("date").result(dt.date)
-        | spaceless_string("time").result(dt.time)
-        | spaceless_string("tinyint", "int1").result(dt.int8)
-        | spaceless_string("integer", "int4", "int", "signed").result(dt.int32)
-        | spaceless_string("ubigint").result(dt.uint64)
-        | spaceless_string("usmallint").result(dt.uint16)
-        | spaceless_string("uinteger").result(dt.uint32)
-        | spaceless_string("utinyint").result(dt.uint8)
-        | spaceless_string("uuid").result(dt.uuid)
-        | spaceless_string("varchar", "char", "bpchar", "text", "string").result(
-            dt.string
+
+SQLGLOT_TYPE_TO_IBIS_TYPE = {
```

Comment:
I'm going to replace our parsy usage in the specifc backends with sqlglot, so this code will be refactored to be more generic then, but for now I'm happy with it as is.

Reply:
Actually I think I'll revert the type parsing changes in this PR and put them in the parsing refactor PR (to-be-PR'd) and use `functools.cache` to speed things up here for the time being.